src/Application/Results.php

```diff
@@ -5,14 +5,15 @@
 namespace JeroenG\Explorer\Application;
 
 use Countable;
+use Elastic\Elasticsearch\Response\Elasticsearch;
 
 class Results implements Countable
 {
     private array $rawResults;
 
-    public function __construct(array $rawResults)
+    public function __construct(array|Elasticsearch $rawResults)
     {
-        $this->rawResults = $rawResults;
+        $this->rawResults = is_array($rawResults) ? $rawResults : $rawResults->asArray();
```

Comment:
I this needed? Isn't it nicer if you call the `Elasticsearch::asArray` on the calling site?

Reply:
I believe this might be going back to the original PR i based this on, happy to make adjustments here as I'm not super tied to anything, we have been using the branch in our production for a few months now. This was just what was required to assure that the tests would pass and the code as originally modified would run.